### PR TITLE
Capture and expose GitHub webhook event GUIDs in more places.

### PR DESCRIPTION
**Pull request by cjwagner**
The event GUID is the value of the `X-Github-Delivery` request header and is used to trace webhook delivery by GitHub internally (and by us).
This PR:
- Makes Gubernator store the event GUID as part of the data it stores for all webhooks and updates Gubernator's timeline page to display the GUIDs next to the event type.
- Makes Prow log receipt of webhooks with unhandled event types and their event GUIDs.

This data will help GitHub support next time we encounter an issue.

/assign @stevekuznetsov @BenTheElder @Katharine

### CODE REVIEW

File: gubernator/github/handlers.py [hunk 1]
```diff
@@ -207,8 +215,13 @@ def emit_events(self, repo, number):
             action = body_json.get('action')
             sender = body_json.get('sender', {}).get('login')
             self.response.write('<tr><td>%s\n' % '<td>'.join(str(x) for x in
-                [event.timestamp, event.event, action, sender,
-                 '<pre>' + cgi.escape(body)]))
+                [   # Table columns
+                    event.timestamp,
+                    '%s<br><pre>%s</pre>' % (event.event, event.guid),
```
Comment (Katharine) [inline gubernator/github/handlers.py]: nit: I'd use `<code>` instead of `<pre>` unless you think this is going to contain significant whitespace (which I'd really hope it does not, given it's supposed to be a guid).

File: gubernator/github/models.py [hunk 2]
```diff
@@ -51,6 +51,7 @@ class GithubWebhookRaw(ndb.Model):
     repo = ndb.StringProperty()
     number = ndb.IntegerProperty(indexed=False)
     event = ndb.StringProperty()
+    guid = ndb.StringProperty()
```
Comment (Katharine) [inline gubernator/github/models.py]: General new-datastore-property comment: if we will never want to look up a record by `guid`, we should specify `indexed=False` on this. It seems plausible to me that we may want to do that someday, so I doubt this needs changing — just checking that you've considered it.

Comment (cjwagner) [reply gubernator/github/models.py]: Yeah, this is intentional. We don't look up records by GUIDs now, but we may want to in the future or while debugging since that is the identifier used for tracing.

File: prow/hook/server.go [hunk 3]
```diff
@@ -165,16 +167,7 @@ func (s *Server) needDemux(eventType, srcRepo string) []plugins.ExternalPlugin {
...
-		if repo == srcRepo {
-			matchesRepo = true
-		}
-		// If repo is an org, we need to compare orgs.
-		if !matchesRepo && !strings.Contains(repo, "/") && repo == srcOrg {
-			matchesRepo = true
-		}
-		// No need to continue if the repos don't match.
-		if !matchesRepo {
+		if repo != srcRepo && repo != srcOrg {
```
Comment (fejta) [inline prow/hook/server.go]: Seems orthogonal?

Comment (cjwagner) [reply prow/hook/server.go]: Yeah it totally is. I was irritated by the unnecessary complexity and figured it was a small enough clean up to include in this PR.

/hold

Comment (cjwagner) [reply prow/hook/server.go]: oops
/hold cancel
